### PR TITLE
[FLINK-35597][test] Fix unstable LocatableSplitAssignerTest#testConcurrentSplitAssignmentForMultipleHosts

### PR DESCRIPTION
**Pull request by liyubin117**
## Brief change log

* To ensure at least one fraction of hosts to be local, it should be `ia.getNumberOfLocalAssignments()` as before.

### CODE REVIEW

**Review by 1996fanrui — APPROVED** [review]
Thanks @liyubin117 for the fix!

LGTM

cc @GOODBOY008 , it's caused by https://github.com/apache/flink/pull/24881

**Comment by liyubin117** [comment]
@1996fanrui @GOODBOY008 Thanks for your attention, I found that CI e2e tests have stucked for hours, As the code is about to be frozen, could you help cancel some tasks? For example, this task has already failed but is still running the e2e test.
https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=60262&view=results

**Comment by 1996fanrui** [comment]
> @1996fanrui @GOODBOY008 Thanks for your attention, I found that CI e2e tests have stucked for hours, As the code is about to be frozen, could you help cancel some tasks? For example, this task has already failed but is still running the e2e test. https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=60262&view=results

No worries, the bugfix still can be merged after feature freeze. The feature cannot be merged after  feature freeze.

Let's wait for the CI for a while.

**Pull request merged**
